### PR TITLE
 Fix income restricted issues 267 and 265 

### PR DESCRIPTION
**Pull request by mbh329**
`income_restricted_units.py`

This indicator was originally using one source data for two separate data points according to our original data matrix. However, we learned that these data points weren't capturing what the SME's were looking for so this pull request addresses the issues as stated in #267 and #265. 

#265 
We needed to change the source data as the old data point `units_rad` was not what the SME's wanted, they were looking for the number of total units coming from the housing new york data. In the script, I loaded in this data and cleaned the data with established conventions and added a new helper function that converts NTA's to PUMAs. I also created a separate accessor function to handle this one data point as it used different data from the original accessor. 

#267 
This was an easy fix and just pointed the `units_nycha` to a the proper column as mentioned in this issue.

`CD_helpers.py`

Added more descriptive name for the get crosswalk, renamed all the columns in the xlsx sheet and then created a new function to go from nta to puma.

### CODE REVIEW

**Comment by mbh329** [comment]
Things to note:
did not add accessor to list @SashaWeinstein 
check column name is correct @AmandaDoyle

**Comment by mbh329** [comment]
@SashaWeinstein the `os` and `numpy` imports are also autoimports I believe? Will they break the tests as well? Ill take a look at that documentation because I know this has happened before

File: internal_review/housing_security/borough/income_restricted_units.csv [hunk 1]
```diff
@@ -1,6 +1,6 @@
-borough,nycha_units_count,rad_units_count
-BK,54681.0,3940.0
-BX,40866.0,2458.0
-MN,52539.0,1718.0
-QN,15478.0,1395.0
-SI,4510.0,0.0
+borough,units_nycha_count
+BK,58621.0
+BX,43324.0
```
[the remark this answers is not in the document]
Comment (td928) [reply internal_review/housing_security/borough/income_restricted_units.csv:3]: I think are due to now the column are basically the sum of the two old columns

File: aggregate/housing_security/income_restricted_units.py [hunk 2]
```diff
@@ -34,9 +41,58 @@ def load_clean_income_restricted():
...
+            [(final, "income_restricted_units_hpd.csv", geography)],
+            "housing_security",
+        )
+    return final
+
+
+def load_clean_hny_data():
+    source_data = pd.read_csv(
+        ".library/hpd_hny_units_by_building.csv",
+        usecols=[
```
[the remark this answers is not in the document]
Comment (mbh329) [reply aggregate/housing_security/income_restricted_units.py:69]: We don't need all these columns but I find these columns are good to help debug any issues (i.e. added the nta column after realizing it geocoded better than the lat/long did)

File: internal_review/housing_security/borough/income_restricted_units.csv (hunk 1 above)
[the remark this answers is not in the document]
Comment (SashaWeinstein) [reply internal_review/housing_security/borough/income_restricted_units.csv:3]: oh yea I see that makes sense

**Review by AmandaDoyle — APPROVED** [review]
- Given that `units_hpd` and `units_nycha` are coming from 2 different sources and going into different output tables, a future enhancement may be to split these into two different scripts.
- Field names `units_nycha` and `units_hpd` are correct.
- In the source data whenever a record has an NTA assigned it also has a lat/long so just using the NTA field should be fine.  Also, given this we shouldn't need to update the method for housing production just yet.

**Comment by mbh329** [comment]
@AmandaDoyle got it

**Comment by mbh329** [comment]
@SashaWeinstein see new commit for requested changes

File: utils/CD_helpers.py [hunk 3]
```diff
@@ -68,14 +68,8 @@ def community_district_to_PUMA(df, CD_col, CD_abbr_type="alpha_borough"):
...
-
-    mapper = {}
-
-    for _, row in puma_cross.iterrows():
-        for nta_num in re.findall(r"\d+", row["nta"]):
-            nta_code = row["nta"][:2] + nta_num
-        mapper[nta_code] = row.puma
-    df["puma"] = df[nta_col].map(mapper)
+    puma_cross = dict(zip(puma_cross.nta, puma_cross.puma))
+    df["puma"] = df[nta_col].map(puma_cross)
```
Comment (SashaWeinstein) [inline utils/CD_helpers.py:72]: nice, glad this worked

**Review by SashaWeinstein — APPROVED** [review]
My comments were addressed, all looks good to me

**Pull request merged**